File: bintree_py/test/testtraversal01.py
from .testbase import display_tree
from binarytree import BinSearchTree


def do_task():
    bst = BinSearchTree()

    for value in [12, 39, 20, 7, 26, 45, 19, 8]:
        bst.insert(value)

    print('\n display tree')
    display_tree(bst)

    print('\n\n in-order traversal:')
    print(bst.traverse('in'))

    print('\n\n pre-order traversal:')
    print(bst.traverse('pre'))

    print('\n\n post-order traversal:')
    print(bst.traverse('post'))

    print()

File: bintree_java/src/test/TestTraversal01.java
package test;


import java.util.List;
import binarytree.BinSearchTree;
import binarytree.traversal.OrderTraversal;



public class TestTraversal01 extends TestBase {

    @Override
    public void doTask() {
        var bst = new BinSearchTree<Integer, BinNodeImp<Integer> >();

        for ( var value : new int[] {12, 39, 20, 7, 26, 45, 19, 8} )
            bst.insert(value);


        System.out.println("\n display tree:");
        displayTree(bst);


        System.out.println("\n\n in-order traversal:");
        printList(bst.traverse(OrderTraversal.IN));

        System.out.println("\n\n pre-order traversal:");
        printList(bst.traverse(OrderTraversal.PRE));

        System.out.println("\n\n post-order traversal:");
        printList(bst.traverse(OrderTraversal.POST));
    }


    private void printList(List<?> lst) {
        for (var value : lst) {
            System.out.print(value + "  ");
        }

        System.out.println();
    }
}
